proto/api.go

```diff
@@ -78,7 +78,7 @@ func IsRange(args Request) bool {
 type Request interface {
 	gogoproto.Message
 	// Header returns the request header.
-	Header() *RequestHeader
+	Header() *KVRequestHeader
```

Comment:
This change is why I think it's better to do the split first instead of committing this rename as the intermediate state. `Request.Header()` should continue to be defined in terms of the common header; we should introduce a new interface `KVRequest` with a `KVHeader` method.